### ISSUE TITLE
Mvn pull from org.scala-lang 2.10.6 not working

### ISSUE THREAD

**Issue by RonaldWentworth** [sbt-compiler-maven-plugin/sbt-compiler-maven-plugin#38]
This build is doesn't compile when included

**Comment by gslowikowski** [comment]
Which build?

**Comment by RonaldWentworth** [comment]
The one that build 12 hrs ago

**Comment by RonaldWentworth** [comment]
we're using the play framework and last nights build failed after this update. We didn't change anything we're getting the following error.
[ERROR] Failed to execute goal com.google.code.sbt-compiler-maven-plugin:sbt-compiler-maven-plugin:1.0.0-beta9:compile (default-compile) on project trucourse-hardware-api: Compiler autodetection failed: Missing:
[ERROR] ----------
[ERROR] 1) com.typesafe.sbt:incremental-compiler:jar:0.13.12
[ERROR] 
[ERROR] Try downloading the file manually from the project website.
[ERROR] 
[ERROR] Then, install it using the command:
[ERROR] mvn install:install-file -DgroupId=com.typesafe.sbt -DartifactId=incremental-compiler -Dversion=0.13.12 -Dpackaging=jar -Dfile=/path/to/file
[ERROR] 
[ERROR] Alternatively, if you host your own repository you can deploy the file there:
[ERROR] mvn deploy:deploy-file -DgroupId=com.typesafe.sbt -DartifactId=incremental-compiler -Dversion=0.13.12 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR] Path to dependency:
[ERROR] 1) dummy:dummy:jar:1.0
[ERROR] 2) com.google.code.sbt-compiler-maven-plugin:sbt-compiler-sbt013:jar:1.0.0-beta9
[ERROR] 3) com.typesafe.zinc:zinc:jar:0.3.12
[ERROR] 4) com.typesafe.sbt:incremental-compiler:jar:0.13.12
[ERROR] 
[ERROR] 2) org.scala-lang:scala-reflect:jar:2.10.6
[ERROR]

**Comment by RonaldWentworth** [comment]
I don't have this in my plugin defined in my dependencies normally.  I added the plugin definition to point to beta8 and it seems to be working.
            <plugin>
                <groupId>com.google.code.sbt-compiler-maven-plugin</groupId>
                <artifactId>sbt-compiler-maven-plugin</artifactId>
                <version>1.0.0-beta8</version>
                <executions>
                    <execution>
                        <id>default-sbt-compile</id>
                        <goals>
                            <goal>compile</goal>
                            <goal>testCompile</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>

**Comment by RonaldWentworth** [comment]
`<plugin>
                <groupId>com.google.code.sbt-compiler-maven-plugin</groupId>
                <artifactId>sbt-compiler-maven-plugin</artifactId>
                <version>1.0.0-beta8</version>
                <executions>
                    <execution>
                        <id>default-sbt-compile</id>
                        <goals>
                            <goal>compile</goal>
                            <goal>testCompile</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>           
`

**Comment by RonaldWentworth** [comment]
If I run
mvn dependency:get -Dartifact=org.scala-lang:scala-compiler:2.10.6
it doesn't find the jar

**Comment by gslowikowski** [comment]
I tried to reproduce your problem, but everything works. BTW in your log I see that Maven cannot get `com.typesafe.sbt:incremental-compiler:0.3.12` (not `org.scala-lang:scala-compiler:2.10.6`). Its available in the [central repo](http://repo1.maven.org/maven2/com/typesafe/sbt/incremental-compiler/0.13.12/).

Can you reproduce this problem locally? Maybe you have some proxy or corporate Maven repository (like Nexus) with some restrictive rules causing problems.

Anyway, try to reproduce this locally. If it would fail, delete `~/.zinc` and `~/.m2/repository/com/typesafe/sbt` directories (backup them somewhere, maybe their content is causing problems for some reason and will be needed for further investigation) and try to build the project again.

**Comment by gslowikowski** [comment]
You can try to add the compiler as plugin's dependency to turn off automatic SBT compiler version discovery, and see, if it helps.

This is described in [documentation](http://sbt-compiler-maven-plugin.github.io/sbt-compiler-maven-plugin/1.0.0-beta9/sbt-compilers/sbt-compiler-sbt013/usage.html) in section `Compiler as plugin dependency`.

**Comment by RonaldWentworth** [comment]
Build 2.10.6 show's that it's there but the jar is not present. I spent most of the day yesterday tracing this down and can see in my .m2 repository that 2.10.6 isn't there. I know this isn't your issue but your change to set to use 2.10.6 by default makes this appear.

**Comment by RonaldWentworth** [comment]
Well looks like they fixed it this morning it's there now.

**Comment by gslowikowski** [comment]
I'm glad, it works for you now.

`1.0.0-beta9` version of `sbt-compiler-maven-plugin` depends on Zinc 0.3.12 which depends on SBT 0.13.12 and it depends on Scala 2.10.6
Previous `1.0.0-beta8` version depends on Zinc 0.3.9 which depends on SBT 0.13.9 and it depends on Scala 2.10.5.
That's why Scala version changed.

But Scala 2.10.6 is available in Maven central repo since 22-Sep-2015 and I used this version for months. SBT 0.13.12 and Zinc 0.3.12 - since 16-Jul-2016 and I tested it for some time before commiting and performing release. Because I've never had problems similar to your, I think it's something related to your corporate network, proxies, corporate repositories, etc.
